vm_setup_vars.yml

```diff
@@ -64,7 +64,7 @@ provisioning_network:
 external_network:
   - name: "{{ baremetal_network_name }}"
     bridge: "{{ baremetal_network_name }}"
-    forward_mode: "{{ 'bridge' if lookup('env', 'MANAGE_BR_BRIDGE') == 'n' else 'nat' if not lookup('env', 'INSTALLER_PROXY') else 'route'}}"
+    forward_mode: "{{ 'bridge' if lookup('env', 'MANAGE_BR_BRIDGE') == 'n' else 'nat' if not (lookup('env', 'INSTALLER_PROXY') or lookup('env', 'DISCONNECTED_ENV')) else 'route'}}"
```

Comment:
Can you just use
`export INSTALLER_PROXY=true`
this is how we have been testing disconnected installs

Reply:
I've tried to use it in [CI](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/42043/rehearse-42043-pull-ci-openshift-appliance-master-e2e-sno-ipv4-dhcp/1688467713762004992), but no luck. Also haven't seen any job in CI already using it.
I'm getting errors as:
```
network.sh: line 7: python: command not found
```
```
+++(/root/dev-scripts/common.sh:8): source(): go env
/root/dev-scripts/common.sh: line 8: go: command not found
++(/root/dev-scripts/common.sh:8): source(): eval ''
/root/dev-scripts/common.sh: line 10: GOPATH: unbound variable
```

But anyway, we don't really need a proxy for the appliance flow. A fully disconnect env should be better for testing.

Reply:
See https://github.com/openshift/release/blob/2f8192da70d24625d1f109a79da75323bc3e0296/ci-operator/step-registry/baremetalds/e2e/proxy/ipv4/baremetalds-e2e-proxy-ipv4-workflow.yaml#L15 for example (the two workflows were initially designed for the metal jobs, not sure about their current state though cc @derekhiggins @tdomnesc )

Reply:
> https://github.com/openshift/release/blob/2f8192da70d24625d1f109a79da75323bc3e0296/ci-operator/step-registry/baremetalds/e2e/proxy/ipv4/baremetalds-e2e-proxy-ipv4-workflow.yaml#L15

@andfasano  Yeah, tried to check it, but no CI test seems to actually use it. Any way, why can't we have a falg for a completely disconnected env? We don't need a proxy at all in the appliance flow.

Reply:
For a real disconnected environment my preference would be to remove at all the forward element, as reported in the libvirt [doc](https://libvirt.org/formatnetwork.html)

Reply:
> I've tried to use it in [CI](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/42043/rehearse-42043-pull-ci-openshift-appliance-master-e2e-sno-ipv4-dhcp/1688467713762004992), but no luck. Also haven't seen any job in CI already using it. I'm getting errors as:
> 
> ```
> network.sh: line 7: python: command not found
> ```
> 
> ```
> +++(/root/dev-scripts/common.sh:8): source(): go env
> /root/dev-scripts/common.sh: line 8: go: command not found
> ++(/root/dev-scripts/common.sh:8): source(): eval ''
> /root/dev-scripts/common.sh: line 10: GOPATH: unbound variable
> ```
> 
> But anyway, we don't really need a proxy for the appliance flow. A fully disconnect env should be better for testing.

Maybe some problem with your dev-scripts setup? Did a quick local test and the network was setup as expected: 

```
<network xmlns:dnsmasq='http://libvirt.org/schemas/network/dnsmasq/1.0'>
  <name>ostestbm</name>
  <uuid>f3b24684-50b4-48e6-b261-ee3706352b0c</uuid>
  <forward mode='route'/>
 ...
```

Reply:
> > I've tried to use it in [CI](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/42043/rehearse-42043-pull-ci-openshift-appliance-master-e2e-sno-ipv4-dhcp/1688467713762004992), but no luck. Also haven't seen any job in CI already using it. I'm getting errors as:
> > ```
> > network.sh: line 7: python: command not found
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```
> > +++(/root/dev-scripts/common.sh:8): source(): go env
> > /root/dev-scripts/common.sh: line 8: go: command not found
> > ++(/root/dev-scripts/common.sh:8): source(): eval ''
> > /root/dev-scripts/common.sh: line 10: GOPATH: unbound variable
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > But anyway, we don't really need a proxy for the appliance flow. A fully disconnect env should be better for testing.
> 
> Maybe some problem with your dev-scripts setup? Did a quick local test and the network was setup as expected:
> 
> ```
> <network xmlns:dnsmasq='http://libvirt.org/schemas/network/dnsmasq/1.0'>
>   <name>ostestbm</name>
>   <uuid>f3b24684-50b4-48e6-b261-ee3706352b0c</uuid>
>   <forward mode='route'/>
>  ...
> ```

Yeah, it works locally. But can't make it work CI.